### ISSUE TITLE
Missing `License-File` in package metadata

### ISSUE THREAD

**Issue by mtelka** [sphinx-doc/sphinx#12713]
### Describe the bug

Packages like `sphinxcontrib_applehelp` are delivering their license text in non-standard and unexpected files (like `LICENCE.rst`).  It would be great if location of the license is added to the package metadata, namely `License-File` (see PEP-639).

### How to Reproduce

Download the `sphinxcontrib_applehelp-2.0.0.tar.gz` sdist from PyPI and unpack it.  Look into the `sphinxcontrib_applehelp-2.0.0/PKG-INFO` file.

### Environment Information

```text
N/A
```

**Comment by AA-Turner** [comment]
> non-standard and unexpected files (like `LICENCE.rst`)

PEP 639 (draft) and Setuptools both suggest a file of the form ``'LICEN[CS]E*'``, which this matches. (Setuptools [goes](https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-license-files) [further](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration) by looking for ``['LICEN[CS]E*', 'COPYING*', 'NOTICE*', 'AUTHORS*']``.

I am happy to accept a PR, though.

A

**Comment by mtelka** [comment]
Yes, the `LICENCE.rst` filename seems to be okay for `setuptools` but it does not look so for `flit-core`.  As a consequence the `sphinxcontrib-applehelp 2.0.0` wheel at PyPI is missing any license file, while the wheel for `sphinxcontrib-applehelp 1.0.8` properly includes the `LICENSE` file.

Re PR: I'm very sorry, but I cannot create one because I've no idea how to specify something like setuptools' `license_files = LICENCE.rst` for `flit-core`.

**Comment by stefan6419846** [comment]
I just stumbled upon this as well. According to https://flit.pypa.io/en/stable/pyproject_toml.html#new-style-metadata, it should just work:

```
license-files

    A list of glob patterns for license files to include. Defaults to ['COPYING*', 'LICEN[CS]E*'].
```

Doing some testing, this is indeed a tricky issue with different configurations conflicting here without a proper message. Using `flit==3.12` gave me the following results:

* Just adding `license-files` explicitly. Error:

  > flit_core.config.ConfigError: license-files cannot be used with a license table, use 'project.license' with a license expression instead

* Replacing `license.text` with `license` as well. Error:

  > flit_core.config.ConfigError: License classifiers are deprecated in favor of the license expression. Remove the 'License :: OSI Approved :: BSD License' classifier

* Removing the classifier as well includes the license files. Now we could remove the `license-files` key again, as the default behavior just works fine now.
